### ISSUE TITLE
QSslSocket: cannot call unresolved function SSLv23_client_method

### ISSUE THREAD

**Issue by nmors** [wkhtmltopdf/wkhtmltopdf#2938]
running on centos6 to a HTTPS address, same binary works on my ubuntu machine

QSslSocket: cannot call unresolved function SSLv23_client_method
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error

wkhtmltopdf exited with code 1 [STACK TRACE]:Error: QSslSocket: cannot call unresolved function SSLv23_client_method
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error

wkhtmltopdf exited with code 1

Note, this is using the generic stable binary "Linux". (which I assume should work on centos) I will see how the "Bleeding Edge" centos version works.

I'll see if compiling on this centos server fixes the issue and close if it does.. maybe the binary is wrong

**Comment by ashkulz** [comment]
@nmors: are you using the latest CentOS 6 release? The generic linux binary **is** CentOS, so you should not bet getting this issue.

**Comment by nmors** [comment]
Yep, **Centos 6.7 (Final)** will ALL latest patches

**Comment by nmors** [comment]
```
# yum install openssl openssl-devel
Loaded plugins: fastestmirror, presto
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: centos.den.host-engine.com
 * epel: archive.linux.duke.edu
 * extras: bay.uchicago.edu
 * updates: bay.uchicago.edu
Package openssl-1.0.1e-42.el6_7.4.x86_64 already installed and latest version
Package openssl-devel-1.0.1e-42.el6_7.4.x86_64 already installed and latest version
Nothing to do
```

**Comment by nmors** [comment]
Okay!

**installing the 32-bit version of openssl fixes the issue.**

This is strange, because I am using the **64 bit version of wkhtmltopdf on a 64-bit system.**

Is it because the SSL request is being made by qt library, and packaged qt is 32 bit ?


Solution:
```
yum install openssl.i686
```

**Comment by ashkulz** [comment]
Does uninstalling the 32-bit OpenSSL cause the issue again? If not, it could be a missing symlink. Can you reproduce this in a fresh VM?

**Comment by basic99** [comment]
QSslSocket: cannot call unresolved function SSLv23_client_method8%
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error
QSslSocket: cannot call unresolved function SSLv23_client_method
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error
QSslSocket: cannot call unresolved function SSLv23_client_method
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error
QSslSocket: cannot call unresolved function SSLv23_client_method
QSslSocket: cannot call unresolved function SSL_CTX_new
QSslSocket: cannot call unresolved function SSL_library_init
QSslSocket: cannot call unresolved function ERR_get_error

I got this and was fixed with yum install openssl.i686 also.
Red Hat Enterprise Linux Server release 6.8 (Santiago)

**Comment by mlutfy** [comment]
For those like me stumbling on this issue, for Debian users, see this comment with a simple fix:  
https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3001#issuecomment-309673453

tl;dr: `apt-get install libssl1.0-dev`

Rather odd, but it works!

**Comment by dunebaud** [comment]
I have the same problem with 0.12.5-dev (builded from source) and  0.12.4 on ubuntu 17.10.
I have tried many solution like:
sudo apt-get install libssl1.0-dev
or
sudo apt-get install libssl1.0.0/artful libssl-dev/artful openssl/artful; sudo apt-get install libssl1.0.0:i386
or
sudo apt-get install libssl1.0.0=1.0.2g-1ubuntu13.4

But none of them worked out.
The only way I could solve my problem was to replace the google font inclusion: https://fonts.googleapis.com/css?family=... by a <style>.download and include the font here..</style> inside my html to be converted...

**Comment by panuci** [comment]
Got the same error, find the problem was when i try to convert "HTML to IMAGE/PDF" and the HTML has images with full url (http://do...).

**Comment by kevindecapite** [comment]
On Ubuntu 16.04.4 LTS, when I run `apt-get install libssl1.0-dev` to try and fix this issue, I get the following:

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libssl1.0-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libssl1.0-dev' has no installation candidate
```

I don't know what the next steps should be.

**Comment by krishjun** [comment]
Thank you @kevindecapite your method worked for me in php-fpm docker container.

**Comment by polaco1782** [comment]
> apt-get install libssl1.0-dev

I confirm this solution works on Debian strech. It is weird, depending on a devel package to work correcly? :confused:
